src/leveled_sst.erl

```diff
@@ -1725,20 +1764,100 @@ hmac(Int) when is_integer(Int) ->
 %%
 %% This implementation of the SlotIndex uses leveled_tree
 
-from_list(SlotList) ->
-    leveled_tree:from_orderedlist(SlotList, ?TREE_TYPE, ?TREE_SIZE).
+from_list(SlotList, FirstKey, LastKey) ->
+    FilterFun = get_filterfun(FirstKey, LastKey),
+    FilteredList =
+        lists:map(fun({K, S}) -> {FilterFun(K), S} end, SlotList),
+    {leveled_tree:from_orderedlist(FilteredList, ?TREE_TYPE, ?TREE_SIZE),
+        FilterFun}.
+
+-spec get_filterfun(
+    leveled_codec:ledger_key(), leveled_codec:ledger_key()) ->
+        fun((leveled_codec:ledger_key())
+            -> leveled_codec:ledger_key()|leveled_codec:slimmed_key()).
+get_filterfun(
+        {Tag, Bucket, {Field, FT}, FK}, {Tag, Bucket, {Field, LT}, LK})
+            when is_binary(FT), is_binary(FK), is_binary(LT), is_binary(LK) ->
+    case {binary:longest_common_prefix([FT, LT]), byte_size(FT)} of
+        {N, M} when N > 0, M > N ->
```

Comment:
I keep looking at this and wondering if there could ever be a situation where FT == LT

Reply:
If there are a lot of keys with exactly the same term on an index, then yes.  That needs a unite test

Reply:
In that case when FT==LT, won't M==N? Then the test M > N is false and this won't use the more optimised FilterFun

Reply:
That is correct.  So it works functionally (as it switches to using {Term, Key} on the index, but is not optimised.  I've added a unit test to show that it at least works.

I don't think changing M > N to M >= N will work, so we would need another clause to optimise.

Reply:
See https://github.com/martinsumner/leveled/pull/383/commits/9db486671cf5b2576ea33fb5d1dbd639c499d7aa